packages/components/Button/src/useButton.ts

```diff
@@ -36,15 +42,19 @@ export const useButton = (props: ButtonProps): ButtonInfo => {
 
   // GH #1336: Set focusRef to null if button is disabled to prevent getting keyboard focus.
   const focusRef = isDisabled ? null : componentRef;
-  const onClickWithFocus = useOnPressWithFocus(focusRef, onClick);
```

Comment:
Seems like it would be less invasive to have the macos version of this not call focus (and maybe rename the function to be something a bit more cross plat friendly) rather than having to remember to fork this behavior on every platform

Reply:
Hooks can't be conditionally called is one issue. The other is we export this hook and there's probably a case where someone wants focus with click. My thought was it would be weird if the hook didn't work for them suddenly and silently.

Reply:
Though I can undo the 'inner' rename here..

Reply:
Could add a wrapper and leave the core function alone if you want.

I think it's likely someone will forget that macOS needs different behavior (or windows needs particular behavior) when adding a new control

Reply:
I did a code search internally, and nobody is directly referencing this method. So I think it'll be safe for me to change the implementation and update docs...